mrp_bom_product_allergen/models/product_product.py

```diff
@@ -0,0 +1,31 @@
+# Copyright 2025 Tecnativa - Christian Ramos
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+
+from odoo import models
+
+
+class ProductProduct(models.Model):
+    _inherit = "product.product"
+
+    def get_bom_lines(self, bom=None):
+        """Get BoM lines recursively for the product."""
+        bom_lines_ids = self.bom_ids.bom_line_ids.ids
```

Comment:
Why using here `ids` instead of records? I suppose it's because what we talked about not iterating and adding records to recordsets, but it's not the same to go record per record, as this case that is recordset per recordset, so I think it's OK to add them as recordsets.